### PR TITLE
Feature/non-se-members

### PR DESCRIPTION
**Pull request by skaramicke**
Things I feel should be refactored separately, or perhaps changed at some point:
- Perhaps rename `URL_EVENTS_API` to something like `URL_REST_API` since it's used for multiple things, or add a copy of the value called `URL_NON_MEMBER_API`, or just leave it.
- We could grab the firstName, lastName and contact_info, so that data is already there for users who manage to log in before we add privacy stuff to non members and let them access the rest of the app. That data is currently only stored when isMember is true, and only upon creation.

Note: This PR also fixes the refresh_token endpoint throwing input validation errors.

### CODE REVIEW

**Review by skaramicke — COMMENTED** [review]
Getting {"status": "INVALID"} from auth API, waiting on help for help from Anders.

**Comment by skaramicke** [comment]
Screenshots obtained by temporarily setting `user["isMember"] = False` in `authm`

Member login form             |  Non member login form
:-------------------------:|:-------------------------:
Simulator Screenshot - iPhone 14 Pro Max (6 7  Display) - 2024-04-29 at 14 46 40 | Simulator Screenshot - iPhone 14 Pro Max (6 7  Display) - 2024-04-29 at 14 46 43

Tab bar contents             |  Settings form
:-------------------------:|:-------------------------:
Simulator Screenshot - iPhone 14 Pro Max (6 7  Display) - 2024-04-29 at 14 46 29 | Simulator Screenshot - iPhone 14 Pro Max (6 7  Display) - 2024-04-29 at 14 46 32

**Comment by Imamyable** [comment]
I actually like URL_EVENTS_API, since we call the hosting site "eventsiten".

File: app/features/common/components/NonMemberInfo.tsx [hunk 1]
```diff
@@ -47,6 +48,7 @@ const NonMemberInfo: React.FC = () => {
                 </HStack>
             </TouchableOpacity>
             <AlertDialog isOpen={showMoreInfo} onClose={() => setShowMoreInfo(false)}>
+                <AlertDialogBackdrop />
```
Comment (skaramicke) [inline app/features/common/components/NonMemberInfo.tsx:51]: <ModalBackdrop bg="$coolGray500" />

**Comment by skaramicke** [comment]
Member login form             |  Non member login form
:-------------------------:|:-------------------------:
simulator_screenshot_84EE5247-4269-49CA-B114-EB9DDAA2753E | simulator_screenshot_57E3FC94-00BC-4A2B-82D4-6E5718379937

**Review by Imamyable — APPROVED** [review]

**Pull request merged**